packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedFieldMessageEmbed.ts

```diff
@@ -166,15 +165,8 @@ export class PaginatedFieldMessageEmbed<T> extends PaginatedMessage {
 		return items.slice(offset, offset + perPageItems);
 	}
 
-	private resolveTemplate(template: EmbedBuilder | EmbedData | ((embed: EmbedBuilder) => EmbedBuilder)) {
-		if (template instanceof EmbedBuilder) {
-			return template;
-		}
-
-		if (isFunction(template)) {
-			return template(new EmbedBuilder());
-		}
-
-		return new EmbedBuilder(template);
+	private resolveTemplate(template: EmbedResolvable | EmbedData | ((embed: EmbedBuilder) => EmbedResolvable)): APIEmbed {
+		if (isFunction(template)) template = template(new EmbedBuilder());
+		return (isJSONEncodable(template) ? template : new EmbedBuilder(template)).toJSON();
```

Comment:
Template should still be a union with `EmbedBuilder`, otherwise this is a breaking change.